pkg/resourcemonitor/podresourcesscanner.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+	"sigs.k8s.io/node-feature-discovery/pkg/apihelper"
+)
+
+type PodResourcesScanner struct {
+	namespace         string
```

Comment:
Shouldn't we watch all namespaces, every time? Otherwise the accounting just can't work

Reply:
I think we discussed this here [here](https://github.com/kubernetes-sigs/node-feature-discovery/pull/525#discussion_r668344868). By default we watch all namespaces. But with this flag we have the ability to monitor specific namespaces in nfd-topology-updater which is useful when we are (unit and E2E) testing this feature. With this capability we don't have to account for resource utilization of all the (infrastructure and application pods) that might have been deployed on the cluster and can check the exact amount of resources decremented based on the resource allocation of pods running in a certain namespace, allowing us not having to account for all the resources consumed by all the pods in the cluster.